exercises/06-redux-saga/src/modules/users/users-saga.ts

```diff
@@ -0,0 +1,42 @@
+import { all, call, fork, put, takeEvery } from 'redux-saga/effects'
+import { Action } from 'redux'
+
+import { usersActions } from 'modules/users/users-slice'
+import * as UsersEffects from 'modules/users/users-effects'
+
+function* getUsers() {
+  try {
+    const { data: users } = yield call(UsersEffects.getUsers)
+    yield put(usersActions.usersLoaded(users))
+  } catch (error) {
+    console.log(error.message)
+  }
+}
+
+function* addUser(action: Action) {
+  if (!usersActions.addUser.match(action)) {
+    console.error(
+      'Unexpected type',
+      `'${action.type}'`,
+      'was passed to addUser saga.'
+    )
+    return
+  }
+
+  const { payload: user } = action
+  try {
+    const response = yield call(UsersEffects.addUser, user)
```

Comment:
When I work with Redux Saga in TS, I explicitly type variables before `yield`. Every result of `yield` is `any`. It doesn't bring any type safety for this line but it might be helpful for the next lines.

```ts
const response: SomeType = yield call(UsersEffects.addUser, user)
```

Reply:
Added types, see 582427bd79610f5521d03d45765387e93c307e67

Reply:
Would it be possible to use some generic to have typing for the `data` field as well? :thinking:

Reply:
Yes, Thank you! See 74af047